grpc/dispatcher.go

```diff
@@ -0,0 +1,86 @@
+package grpc
+
+import (
+	"context"
+	"time"
+
+	"github.com/gojek/fiber"
+	fiberError "github.com/gojek/fiber/errors"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/proto"
+)
+
+const (
+	TimeoutDefault = time.Second
+)
+
+type Dispatcher struct {
+	Timeout time.Duration
+}
+
+func (d *Dispatcher) Do(request fiber.Request) fiber.Response {
+
+	grpcRequest, ok := request.(*Request)
+	if !ok {
+		return fiber.NewErrorResponse(
+			fiberError.FiberError{
+				Code:    int(codes.InvalidArgument),
+				Message: "fiber: grpc.Dispatcher supports only grpc.Request type of requests",
+			})
+	}
+
+	if grpcRequest.hostport == "" || grpcRequest.ServiceMethod == "" {
+		return fiber.NewErrorResponse(
+			fiberError.FiberError{
+				Code:    int(codes.InvalidArgument),
+				Message: "missing hostport/servicemethod",
+			})
+	}
+
+	conn, err := grpc.Dial(grpcRequest.hostport, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		// if ok is false, unknown codes.Unknown and Status msg is returned in Status
+		responseStatus, _ := status.FromError(err)
+		return &Response{Status: *responseStatus}
+	}
+
+	payload, ok := request.Payload().(proto.Message)
```

Comment:
Is this line necessary? you have converted the request to `grpc.Request` and technically you are able to get the request payload by simply calling `grpcRequest.RequestPayload` in line 70.

```
	err = conn.Invoke(ctx, grpcRequest.ServiceMethod, grpcRequest.RequestPayload, responseProto, grpc.Header(&responseHeader))

```

Reply:
ah nice catch, i didnt realise the signature of the invoke takes in an interface (although it will assert proto.message deeper later)

will remove this chunk